### PR TITLE
[Security/Extension] Role encryption/decryption

### PR DESCRIPTION
**Pull request by RyanL1997**
### Description
_*This idea is coming from @cwperks 's [comment](https://github.com/opensearch-project/security/pull/2567#discussion_r1153350018)*_

Including encrypted `roles` in JIT token as payload for privilege validation.

### Issues Resolved
* Relate: #2573 
* Relate: https://github.com/opensearch-project/security/issues/2619

Is this a backport? If so, please add backport PR # and/or commits #

### Check List
- [x] New functionality includes testing
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java [hunk 1]
```diff
@@ -63,18 +73,20 @@ public void testCreateJwt() throws Exception {
         Assert.assertNotNull(jwt.getClaim("iat"));
         Assert.assertNotNull(jwt.getClaim("exp"));
         Assert.assertEquals(expectedExp, jwt.getClaim("exp"));
+        Assert.assertEquals(expectedRoles, EncryptionDecryptionUtil.decrypt(claimsEncryptionKey, jwt.getClaim("roles").toString()));
```
Comment (cwperks) [inline src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java:76]: Do you think we should add an additional assertion in this test?

```
Assert.assertNotEquals(expectedRoles, jwt.getClaim("roles"));
```

Explicitly asserting that the claim in the payload is encrypted and not the value that original value.

**Review by cwperks — COMMENTED** [review]
@RyanL1997 Thank you for putting a PR out for review for encrypting sensitive claims in the JWT! I'm curious to know what other maintainers think. This looks good to me.

File: src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java (hunk 1 above)
Comment (RyanL1997) [reply src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java:76]: Good catch! I will add that.

**Comment by RyanL1997** [comment]
@opensearch-project/security Hi folks, I may need your opinions on this feature branch testings. For merging this PR, should I cherry pick all the changes from `main` to fix these issues such as, CVE from whitesource security check? Just let me know, if that is the case, I do have a cherry pick PR ready in my fork. :)

**Comment by cwperks** [comment]
@RyanL1997 For the feature branch you will need to sync it periodically to keep it from drifting from main over time. Similar to what was done on `feature/identity` in core. See those PRs here: https://github.com/opensearch-project/OpenSearch/pulls?q=is%3Apr+is%3Aclosed+merge+remote+into+feature%2Fidentity+

File: src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java [hunk 2]
```diff
@@ -154,7 +158,8 @@ public String createJwt(String issuer, String subject, String audience, Integer
             throw new Exception("The expiration time should be a positive integer");
         }
 
-        // TODO: Should call preparelaims() if we need roles in claim;
+        String listOfRoles = String.join(",", roles);
```
Comment (scrawfor99) [inline src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java:161]: If roles is empty, will this throw an uncaught error?

File: src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java [hunk 3]
```diff
@@ -42,17 +44,20 @@ public void testCreateJwkFromSettingsWithoutSigningKey() throws Exception{
     }
```
Comment (scrawfor99) [inline src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java:45]: Maybe add a test where roles is empty or where we test incorrect keys?

**Comment by RyanL1997** [comment]
> @RyanL1997 For the feature branch you will need to sync it periodically to keep it from drifting from main over time. Similar to what was done on `feature/identity` in core. See those PRs here: https://github.com/opensearch-project/OpenSearch/pulls?q=is%3Apr+is%3Aclosed+merge+remote+into+feature%2Fidentity+

Hi @cwperks, thanks for the info. I will cherry pick the changes right away.

File: src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java (hunk 3 above)
Comment (RyanL1997) [reply src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java:45]: done~

File: src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java (hunk 2 above)
Comment (RyanL1997) [reply src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java]: I just updated that.

**Review by scrawfor99 — APPROVED** [review]
Looks good to me! Thanks Ryan.
